src/include/migraphx/generate.hpp

```diff
@@ -36,36 +36,38 @@ inline namespace MIGRAPHX_INLINE_NS {
 template <class T, MIGRAPHX_REQUIRES(is_floating_point<T>{})>
 constexpr T normalize(unsigned long z)
 {
-    if(z == 0)
-        return T(0);
-    const auto max     = 32;
-    const double range = max / 2; // NOLINT
-    double result      = double(z % max) / range;
-    result -= 1;
+    const auto max     = 1ULL << (sizeof(T) * 8 - 1);
+    const double range = max / 2.0;
+    double result      = -1.0 + (z % max) / range;
     return T(result);
+    // Expected output: between -1.0 and 1.0
```

Comment:
Just a minor style issue, but can you move the comment above the return statemen?

Reply:
Of course. Thanks.